Runner/Utils.py

```diff
@@ -0,0 +1,300 @@
+        # Grab all the tasks in the Job.
+        tasks = batch_service_client.task.list(job_id)
+        # tasks = yield loop.run_in_executor(None,
+        # functools.partial(batch_service_client.task.list,
+        # batch_service_client, job_id))
+
+        # Check to see how many tasks are incomplete.
+        incomplete_tasks = [task for task in tasks if
+                            task.state != batchmodels.TaskState.completed]
+        # if the all the tasks are complete we return a complete message, else
+        # we wait all the tasks are complete
+        if not incomplete_tasks:
+            return JobStatus(JobState.COMPLETE,
+                             "Job {} successfully completed.".format(job_id))
+        else:
+            print("Job [{}] is running".format(job_id))
+            time.sleep(10)
+
+    return JobStatus(JobState.NOT_COMPLETE,
+                     "ERROR: Tasks did not reach 'Completed' state within timeout period of: " + str(timeout))
+
+
+def check_task_output(batch_service_client, job_id, expected_output):
+    """Prints the stdout.txt file for each task in the job.
+
+    :param batch_client: The batch client to use.
+    :type batch_client: `batchserviceclient.BatchServiceClient`
+    :param str job_id: The id of the job with task output files to print.
+    """
+
+    tasks = batch_service_client.task.list(job_id)
+
+    for task in tasks:
+        all_files = batch_service_client.file.list_from_task(
+            job_id, task.id, recursive=True)
+
+        for f in all_files:
+            if expected_output in f.name:
+                logger.info(
+                    "Job [{}] expected output matched {}".format(
+                        job_id, expected_output))
+                return JobStatus(JobState.COMPLETE,
+                                 "File found {0}".format(expected_output))
+
+    return JobStatus(JobState.UNEXPECTED_OUTPUT, ValueError(
+        "Error: Cannot find file {} in job {}".format(expected_output, job_id)))
+
+
+def print_result(job_managers):
+    logger.info("Number of jobs run {}.".format(len(job_managers)))
+    failedJobs = 0
+    for i in job_managers:
+        if i.job_status.job_state != JobState.COMPLETE:
+            failedJobs += 1
+            logger.info(
+                "job {} failed because {} : {}".format(
+                    i.job_id,
+                    i.job_status.job_state,
+                    i.job_status.message))
+
+    if failedJobs == 0:
+        logger.info("-----------------------------------------")
+        logger.info("All jobs were successful Run")
+    else:
+        logger.info("-----------------------------------------")
+        logger.info("Number of jobs passed {} out of {}.".format(
+            len(job_managers) - failedJobs, len(job_managers)))
+
+
+def export_result(job_managers, total_time):
+    failedJobs = 0
+    logger.info("Exporting test output file")
+    root = Element('testsuite')
+
+    for i in job_managers:
```

Comment:
**recommendation** rename `i` to more logical contextual name. might be : `job_item`.

Reply:
Yep, i'll change it into job_item